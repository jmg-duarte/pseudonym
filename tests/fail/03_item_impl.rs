use pseudonym::alias;

#[alias(S1, S2)]
struct S(i32);

#[alias(
    S1,
    deprecated(S2),
    deprecated(S3, since = "0.1.0"),
    deprecated(S4, note = "deprecation note"),
    deprecated(S5, since = "0.1.0", note = "deprecation note"),
)]
impl S {
    fn new() -> Self {
        Self(5)
    }
}

fn main() {
    assert_eq!(S::new().0, 5);
    assert_eq!(S1::new().0, 5);
    assert_eq!(S2::new().0, 5);
    assert_eq!(S3::new().0, 5);
    assert_eq!(S4::new().0, 5);
    assert_eq!(S5::new().0, 5);
}