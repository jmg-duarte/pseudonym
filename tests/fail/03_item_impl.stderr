error: unexpected end of input, expected identifier
 --> tests/fail/03_item_impl.rs:6:1
  |
6 | #[alias(S1, S2,)]
  | ^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `alias` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no function or associated item named `new` found for struct `S` in the current scope
  --> tests/fail/03_item_impl.rs:14:19
   |
4  | struct S(i32);
   | -------------- function or associated item `new` not found for this
...
14 |     assert_eq!(S::new().0, 5);
   |                   ^^^ function or associated item not found in `S`

error[E0599]: no function or associated item named `new` found for struct `S1` in the current scope
  --> tests/fail/03_item_impl.rs:15:20
   |
4  | struct S(i32);
   | -------------- function or associated item `new` not found for this
...
15 |     assert_eq!(S1::new().0, 5);
   |                    ^^^ function or associated item not found in `S1`

error[E0599]: no function or associated item named `new` found for struct `S2` in the current scope
  --> tests/fail/03_item_impl.rs:16:20
   |
4  | struct S(i32);
   | -------------- function or associated item `new` not found for this
...
16 |     assert_eq!(S2::new().0, 5);
   |                    ^^^ function or associated item not found in `S2`
